source/fundamentals/builders/filters.txt

```diff
@@ -11,3 +11,684 @@ Filters Builders
+      collection.find(textComparison).into(results);
+      for (Document result : results) {
+         System.out.println(result.toJson());
+      }
+
+   The Bson renders as:
+
+   .. code-block:: none
+      :copyable: false
+   
+      { 
+         "text" : "green"
+      }
+
+   The output of the above code snippet should look something like this:
+   
+   .. code-block:: json
+      :copyable: false
+      
+      {"_id": 7, "color": "green", "qty": 8,"vendor": ["B", "C"]}
+
+   Available with MongoDB 3.2, a version 3 text index allows case-sensitive
+   and diacritic-sensitive searches.
+
+   .. code-block:: java
+
+      text("green", TextSearchOptions().caseSensitive(true))
+
+   .. code-block:: java
+
+      text("green", TextSearchOptions().diacriticSensitive(true))
+
+.. _bitwise:
+
+Bitwise
+-------
+
+The bitwise query operators include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Logical Operator
+     - Selects documents where
+      
+   * - ``bitsAllSet()``
+     - the specified bits of a field are set (i.e. 1).
+
+   * - ``bitsAllClear()``
+     - the specified bits of a field are clear (i.e. 0).
+      
+   * - ``bitsAnySet()``
+     - at least one of the specified bits of a field are set (i.e. 1).
+     
+   * - ``bitsAnyClear()``
+     - at least one of the specified bits of a field are clear (i.e. 0).
+
+The following code snippet shows how to use a bitwise filter:
+
+.. code-block:: java
+   
+   import static com.mongodb.client.model.Filters.*;
+
+   // <MongoCollection setup code here>
+
+   collection.find(<bitwise filter>);
+
+.. example::
+
+   The following example selects documents that have a bitField field
+   with bits set at positions of the corresponding bitmask 50 (i.e.
+   00110010) in the binary_numbers collection: 
+
+   .. code-block:: java
+      
+      import static com.mongodb.client.model.Filters.bitsAllSet;
+
+      // <MongoCollection setup code here>
+
+      Bson bitsComparison = bitsAllSet(1, 5);
+      List<Document> results = new ArrayList<>();
+      collection.find(bitsComparison).into(results);
+      for (Document result : results) {
+         System.out.println(result.toJson());
+      }
+
+   The Bson renders as:
+
+   .. code-block:: none
+      :copyable: false
+   
+      { 
+         "a": 
+            { 
+               "$bitsAllSet": [ 1, 5 ] 
+            } 
+      }
+
+   The output of the above code snippet should look something like this:
+   
+   .. code-block:: json
+      :copyable: false
+      
+      { _id: 9, a: 54, binaryValue: "00110110" }
+      { _id: 12, a: 102, binaryValue: "01100110" }
+
+.. _geospatial:
+
+Geospatial
+----------
+
+The geospatial operator methods include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Comparison Operator
+     - Matches
+      
+   * - ``geoWithin()``
+     - Selects documents containing a GeoJSON geometry field that falls within within a bounding GeoJSON geometry.
```

Comment:
Since we use "matches" in the column header, this reads as "Matches selects documents..." Could you update this for correctness and consistency?